### PR TITLE
chore: build on actions

### PR DESCRIPTION
**Pull request by sfoster1**
We should move this build over to github actions on our self-hosted runners so that we can cut our release stuff over to it.

Annoying note: you can't run `workflow_dispatch` of a workflow if it's not on the main branch, so we can't really properly test this until we merge it. I'm going to add some push triggers just to get the syntax checked first though.

### CODE REVIEW

File: .github/actions/build-refs/action.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+description: "Based on a subset of provided refs for an openembedded build, decide the rest."
+inputs:
+  token:
+    description: "gh token for api calls to resolve refs"
+    required: true
+  monorepo:
+    description: "ref of https://github.com/opentrons/opentrons to build"
+    required: false
+    default: "-"
+  oe-core:
```
Comment (vegano1) [inline .github/actions/build-refs/action.yml]: oe core?

File: .github/actions/build-refs/action/__tests__/index.test.ts [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    'develop',
+  ],
+  [
+    'when monorepo ref is some branch but is not a tag is develop',
+    ['refs/heads/something'],
+    'develop',
+  ],
+  [
+    'when monorepo ref is a release tag is release',
+    ['refs/tags/ot3@0.0.0-dev'],
```
Comment (vegano1) [inline .github/actions/build-refs/action/__tests__/index.test.ts]: I don't think we have a prefix for ot2 builds but this should ot2.

File: .github/actions/build-refs/action/index.ts [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+      await Promise.all(refList.map(ref => refResolves(repo, ref))).then(
+        presentRefs => presentRefs.find(maybeRef => maybeRef !== null)
+      )
+    )
+  }
+  return resolved
+}
+
+export function resolveBuildType(ref: Ref): BuildType {
+  return ref.includes('refs/tags/ot3') ? 'release' : 'develop'
```
Comment (vegano1) [inline .github/actions/build-refs/action/index.ts]: same here, should be for ot2

Comment (sfoster1) [reply .github/actions/build-refs/action/index.ts]: good calls

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        shell: bash
+        id: 'upload-results'
+        run: |
+          pushd buildroot/output/images/
+          aws --profile=${{ steps.aws.outputs.profile_name }} s3 cp --acl=public-read ./* s3://${S3_ARTIFACT_ARN/arn:aws:s3:::/}/ot2-br/${{github.run_id}}
+          root_url=https://${S3_ARTIFACT_ARN/arn:aws:s3:::/}/ot2-br/${{github.run_id}}
+          echo "console_url=https://s3.console.aws.amazon.com/s3/buckets/${S3_ARTIFACT_ARN/arn:aws:s3::::/}?prefix=${{github.run_id}}" >> $GITHUB_OUTPUT
+          echo "version_file_url=$root_url/VERSION.json" >> $GITHUB_OUTPUT
+          echo "release_notes_file_url=$root_url/release-notes.md" >> $GITHUB_OUTPUT
+          echo "system_url=$root_url/ot3-system.zip" >> $GITHUB_OUTPUT
```
Comment (vegano1) [inline .github/workflows/build.yml]: should also set this to ot2-system.zip and ot2-fullimage.zip

Comment (sfoster1) [reply .github/workflows/build.yml]: `ot2-fullimage.zip` even

**Review by vegano1 — APPROVED** [review]

**Pull request merged**